### DISCUSSION TITLE
It will be great if we can use generated class as rootClass

### DISCUSSION THREAD

**Discussion by WestFarmer** [mybatis/generator#1160] [Ideas]
![Uploading 图片.png…]()

**Comment by jeffgbutler** [comment]
I don't understand this idea. Can you elaborate?

**Reply by AlwaysNoobCoder** [reply]
@jeffgbutler 

for example:

```
A.java generated 
B.java generated
```

if I want `B extends A`, it's not possible for current Mybatis generator.

because Mybatis generator need A in java class-path, to analyse fields/methods of A and B, to determine a member belongs to B or A. while at this point, A is not compiled yet, so it's not possible.

while information of A is actually fully accessible to mybatis generator, so I guess maybe  when analysing root class, we should not consider class-path only.

sorry for late reply and poor English, I hope you can understand.
